src/Controls/src/Core/Interactivity/TriggerBase.cs

```diff
@@ -2,25 +2,26 @@
 using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.Runtime.CompilerServices;
 using Microsoft.Maui.Controls.Internals;
 
 namespace Microsoft.Maui.Controls
 {
 	/// <include file="../../../docs/Microsoft.Maui.Controls/TriggerBase.xml" path="Type[@FullName='Microsoft.Maui.Controls.TriggerBase']/Docs/*" />
 	public abstract class TriggerBase : BindableObject, IAttachedObject
 	{
+		static readonly BindableProperty TriggerBaseSpecificityCountProperty = BindableProperty.CreateAttached("TriggerBaseSpecificityCount", typeof(ushort), typeof(TriggerBase), (ushort)0);
+		
+		ConditionalWeakTable<BindableObject, object> _triggerSpecificity = new ConditionalWeakTable<BindableObject, object>();
+		
 		bool _isSealed;
 
-		//each trigger is different
-		static int count = 1;
```

Comment:
If you think of a complex application, where trigger are potentially defined at component level, we might end up adding +1 here many times.
Think for example if we have these components inside a collection view (+ virtualization) with a lot of elements.
We might reach the limit sooner or later, while storing the count of triggers on each `Bindable` makes it safer.